### PR TITLE
Drop Python 3.4 support and update CI

### PR DESCRIPTION
**Pull request by cdeil**
This pull request is a continuation of #686 .

Changes:
- [x] Drop support for Python 3.4 (reason: annoying in CI tests, because 
- [x] Add more Python 3.6 builds in CI, especially Python 3 builds with Sherpa which we didn't have so far in CI.
- [x] Fix some Sphinx errors (that were unrelated, in master)

### CODE REVIEW

**Comment by cdeil** [comment]
Last build in master before this PR:
* Build matrix:
    * https://travis-ci.org/gammapy/gammapy/builds/228097272
    * 20 builds
    * Total time: 3 hrs 2 min 10 sec
* Main build (that's used for coverage) is Python 2.7 with Sherpa 4.9.0:
    * https://travis-ci.org/gammapy/gammapy/jobs/228097278
    * 651 passed, 50 skipped, 26 xfailed, 9 xpassed in 460.89 seconds
    * https://coveralls.io/builds/11340799
    * 0.73 hits per line, 13314 of 18262 relevant lines covered (72.91%)

With the changes from this PR:
* Build matrix
    * https://travis-ci.org/gammapy/gammapy/builds/228110564
    * 20 builds
    * Total time:  3 hrs 20 min 23 sec
* Main build (that's used for coverage) is Python 3.6 with Sherpa 4.9.0:
    * https://travis-ci.org/gammapy/gammapy/jobs/228110570
    * 651 passed, 50 skipped, 26 xfailed, 9 xpassed in 422.94 seconds
    * https://coveralls.io/builds/11341564
    * 0.73 hits per line, 13314 of 18262 relevant lines covered (72.91%)

In short: everything looks fine.

(except for https://travis-ci.org/gammapy/gammapy/jobs/228110585#L2003, for which I'll open a new issue)
